grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -1140,110 +1148,441 @@ class StockMovementService {
         }
     }
 
-    void validatePicklistListImport(StockMovement stockMovement, List<PickPageItem> pickPageItems, List<ImportPickCommand> picklistItems) {
-        Map<String, List> picklistItemsGroupedByRequisitionItem = picklistItems.groupBy { it.id }
+    /**
+     * This method needs to be broken down into multiple smaller methods.
```

Comment:
I agree :laughing:

Reply:
I think you have a comment in the controller class that is alluding to this, but you could have a PicklistImportValidator (that extends some generic ImportValidator) that wraps all this logic and returns a PicklistImportValidationResult object (that extends some generic ValidationResult).

This is a really complex validation step so I don't know exactly what that would look like, but you could break it down further into smaller validators if you really wanted to and just bubble up the ValidationResult.

```
    PicklistImportValidationResult result = picklistImportValidator.validate()
    if (!result.isValid()) {
        result.errors()....
    }
```

And in the PicklistImportValidator:
```
    PicklistImportValidationResult validate() {
       PicklistImportValidationResult result = new()
        ...
        ValidationResult xResult = someMoreSpecificValidator.validate()
        result.addAll(xResult.errors())
    }
```

Again, this is super complex so I don't know how well that would work, but as is, this feels like it really needs to be broken down somehow. The method is pretty unweildly.